### PR TITLE
Css layer refactor

### PR DESCRIPTION
**Pull request by KilDesu**
## PR Type

> What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## What's new?

> List the changes

- Remove all `:global` selectors
- Use cascade layers to handle style scopes
- Add a central disabled state styling

## Screenshots

> If needed, you can add screenshots here

N/A

## This pull request closes an issue

> Add `Closes`, `Fixes` or `Resolves` followed by `#xxx[,#xxx]` where "xxx" is the issue number

N/A

### CODE REVIEW

File: src/lib/components/chip/QChip.svelte [hunk 1]
```diff
@@ -98,6 +98,7 @@
   {...Q.classes}
   aria-disabled={disabled || undefined}
   {tabindex}
+  role="button"
```
Comment (therealcoder1337) [inline src/lib/components/chip/QChip.svelte:101]: Should it always have this?

File: src/lib/css/_disabled.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  @return mixins.add-transparency(var(--#{$color}), 38%);
+}
+
+@function disabled-rest($color: on-surface) {
+  @return mixins.add-transparency(var(--#{$color}), 12%);
+}
+
+[disabled],
+[aria-disabled] {
+  &:not(.q-switch),
```
Comment (therealcoder1337) [inline src/lib/css/_disabled.scss:22]: Component specific stuff should be in the component's `scss`, not here.

File: src/lib/css/index.scss [hunk 3]
```diff
@@ -1,32 +1,23 @@
...
+  }
 
-html,
-body {
-  height: 100%;
-  width: 100%;
-  color: var(--on-surface);
-}
+  @layer components {
+    @import "components";
```
Comment (therealcoder1337) [inline src/lib/css/index.scss:13]: Tree shakability might not work here, but it's something we don't need to consider for this PR.

**Review by therealcoder1337 — CHANGES_REQUESTED** [review]
Great refactor overall, simplifying css. Just a few comments.

File: src/lib/components/chip/QChip.svelte (hunk 1 above)
Comment (KilDesu) [reply src/lib/components/chip/QChip.svelte:101]: Not sure, have to read about this in official ally docs. ImportaProbably not super important atm

File: src/lib/css/theme/reset.scss [hunk 4]
```diff
@@ -9,10 +9,26 @@
...
+
+.body--dark {
+  background-color: rgb(20, 19, 20);
+}
+
 body {
-  color: var(--on-background);
   background-color: var(--body);
   overflow-x: hidden;
+  height: 100%;
```
Comment (therealcoder1337) [inline src/lib/css/theme/reset.scss:30]: Why?

**Review by therealcoder1337 — APPROVED** [review]
Thanks!

File: src/lib/css/theme/reset.scss (hunk 4 above)
Comment (KilDesu) [reply src/lib/css/theme/reset.scss:30]: We'll probably have to change it later but not in this PR. We rven have duplicate styles in this file so we need to be sure of what styles we want to reset.

**Pull request merged**